strawberry/schema/schema_converter.py

```diff
@@ -110,23 +110,20 @@ def from_argument(self, argument: StrawberryArgument) -> GraphQLArgument:
             description=argument.description,
         )
 
-    def from_enum(self, enum: EnumDefinition) -> GraphQLEnumType:
-
-        assert enum.name is not None
-
+    def from_enum(self, enum: StrawberryEnum) -> GraphQLEnumType:
         # Don't reevaluate known types
-        if enum.name in self.type_map:
-            graphql_enum = self.type_map[enum.name].implementation
+        if enum.graphql_name in self.type_map:
+            graphql_enum = self.type_map[enum.graphql_name].implementation
```

Comment:
I need to check if types are unique globally or if they are unique by type (ie I can have a scalar called User and an Enum called User too)